opencti-platform/opencti-front/src/components/FilterIconButtonContainer.tsx

```diff
@@ -0,0 +1,187 @@
+import { last } from 'ramda';
+import Chip from '@mui/material/Chip';
+import Tooltip from '@mui/material/Tooltip';
+import React, { FunctionComponent } from 'react';
+import makeStyles from '@mui/styles/makeStyles';
+import { PreloadedQuery, usePreloadedQuery } from 'react-relay';
+import { truncate } from '../utils/String';
+import { DataColumns } from './list_lines';
+import { useFormatter } from './i18n';
+import { Theme } from './Theme';
+import { Filter } from '../utils/filters/filtersUtils';
+import FilterIconButtonContent, { filterIconButtonContentQuery } from './FilterIconButtonContent';
+import { FilterIconButtonContentQuery } from './__generated__/FilterIconButtonContentQuery.graphql';
+
+const useStyles = makeStyles<Theme>((theme) => ({
+  filter1: {
+    marginRight: 10,
+    lineHeight: 32,
+    marginBottom: 10,
+  },
+  filter2: {
+    margin: '0 10px 10px 0',
+    lineHeight: 32,
+  },
+  filter3: {
+    fontSize: 12,
+    height: 20,
+    marginRight: 7,
+    borderRadius: 10,
+    lineHeight: 32,
+  },
+  operator1: {
+    fontFamily: 'Consolas, monaco, monospace',
+    backgroundColor: theme.palette.background.accent,
+    marginRight: 10,
+    marginBottom: 10,
+  },
+  operator2: {
+    fontFamily: 'Consolas, monaco, monospace',
+    backgroundColor: theme.palette.background.accent,
+    margin: '0 10px 10px 0',
+  },
+  operator3: {
+    fontFamily: 'Consolas, monaco, monospace',
+    backgroundColor: theme.palette.background.accent,
+    height: 20,
+    marginRight: 10,
+  },
+  inlineOperator: {
+    display: 'inline-block',
+    height: '100%',
+    borderRadius: 0,
+    margin: '0 5px 0 5px',
+    padding: '0 5px 0 5px',
+    backgroundColor: 'rgba(255, 255, 255, .1)',
+    fontFamily: 'Consolas, monaco, monospace',
+  },
+  chipLabel: {
+    lineHeight: '32px',
+    maxWidth: 400,
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+}));
+
+interface FilterIconButtonContainerProps {
+  displayedFilters: Filter[];
+  globalMode: string;
+  handleRemoveFilter?: (key: string, op?: string) => void;
+  handleSwitchGlobalMode?: () => void;
+  handleSwitchLocalMode?: (filter: Filter) => void;
```

Comment:
Maybe : handleSwitchFilterOpMode

Reply:
maybe we can have a single hook for handleSwitchGlobalMode and handleSwitchLocalMode, the hook will know in which case we are according if we pass a filter argument or not

Reply:
Yes, it's a better approach